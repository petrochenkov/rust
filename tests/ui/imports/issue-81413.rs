pub const ITEM: Item = Item;

pub struct Item;

pub fn item() {}

pub use doesnt_exist::*;
//~^ ERROR unresolved import `doesnt_exist`
mod a {
    use crate::{item, Item, ITEM};
    //~^ ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
}

mod b {
    use crate::item;
    //~^ ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
    use crate::Item;
    //~^ ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
    use crate::ITEM;
    //~^ ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
}

mod c {
    use crate::item;
    //~^ ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
}

fn main() {}
