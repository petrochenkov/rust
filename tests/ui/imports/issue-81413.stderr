error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:10:17
   |
LL |     use crate::{item, Item, ITEM};
   |                 ^^^^

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:10:17
   |
LL |     use crate::{item, Item, ITEM};
   |                 ^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:10:23
   |
LL |     use crate::{item, Item, ITEM};
   |                       ^^^^

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:10:23
   |
LL |     use crate::{item, Item, ITEM};
   |                       ^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:10:23
   |
LL |     use crate::{item, Item, ITEM};
   |                       ^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:10:29
   |
LL |     use crate::{item, Item, ITEM};
   |                             ^^^^

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:10:29
   |
LL |     use crate::{item, Item, ITEM};
   |                             ^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:21:9
   |
LL |     use crate::item;
   |         ^^^^^^^^^^^

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:21:9
   |
LL |     use crate::item;
   |         ^^^^^^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:24:9
   |
LL |     use crate::Item;
   |         ^^^^^^^^^^^

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:24:9
   |
LL |     use crate::Item;
   |         ^^^^^^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:24:9
   |
LL |     use crate::Item;
   |         ^^^^^^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:28:9
   |
LL |     use crate::ITEM;
   |         ^^^^^^^^^^^

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:28:9
   |
LL |     use crate::ITEM;
   |         ^^^^^^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:34:9
   |
LL |     use crate::item;
   |         ^^^^^^^^^^^

error: cannot determine resolution for the import
  --> $DIR/issue-81413.rs:34:9
   |
LL |     use crate::item;
   |         ^^^^^^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error[E0432]: unresolved import `doesnt_exist`
  --> $DIR/issue-81413.rs:7:9
   |
LL | pub use doesnt_exist::*;
   |         ^^^^^^^^^^^^ use of unresolved module or unlinked crate `doesnt_exist`
   |
help: you might be missing a crate named `doesnt_exist`, add it to your project and import it in your code
   |
LL + extern crate doesnt_exist;
   |

error: aborting due to 17 previous errors

For more information about this error, try `rustc --explain E0432`.
